packages/core/test/controls/numericInputTests.tsx

```diff
@@ -432,6 +467,30 @@ describe("<NumericInput>", () => {
                 const newValue = component.state().value;
                 expect(newValue).to.equal(MAX_VALUE.toString());
             });
+
+            it("fires onValueChange with clamped value if nextProps.max < value ", () => {
+                const onValueChangeSpy = sinon.spy();
+                const component = mount(<NumericInput value={10} onValueChange={onValueChangeSpy} />);
+
+                component.setProps({ max: 0 });
+
+                const newValue = component.state().value;
+                expect(newValue).to.equal("0");
+                expect(onValueChangeSpy.calledOnce).to.be.true;
+                expect(onValueChangeSpy.firstCall.args).to.deep.equal([0, "0"]);
+            });
+
+            it("fires onValueChange with unchanged value if nextProps.max > props.max", () => {
```

Comment:
wait why is it fired at all if the value didn't actually change? this sounds like what we just went through with `EditableText` #629